FBSnapshotTestCase/FBSnapshotTestController.m

```diff
@@ -226,6 +228,15 @@ - (NSString *)_fileNameForSelector:(SEL)selector
   if (0 < identifier.length) {
     fileName = [fileName stringByAppendingFormat:@"_%@", identifier];
   }
+  
+  if (self.isDeviceAgnostic) {
+    UIDevice *device = [UIDevice currentDevice];
+    NSString *model  = [device.model stringByReplacingOccurrencesOfString:@" " withString:@"_"];
+    NSString *os     = [device.systemVersion stringByReplacingOccurrencesOfString:@"." withString:@"_"];
+
+    fileName         = [NSString stringWithFormat:@"%@_%@%@", fileName, model, os];
```

Comment:
Also, were you to separate this logic out into a separate function or object, whose only responsibility was the format the strings returned by `-[UIDevice model]` and `-[UIDevice syetemVersion]`, this code would be a little cleaner. It'd also be easier to test.

Speaking of which, how about some tests for this feature?